DatabaseTypeRecognizer.getDatabaseType(final DataSource dataSource) return null when SQLException happen

## Bug Report

**For English only**, other languages will not accept.

Before report a bug, make sure you have:

- Searched open and closed [GitHub issues](https://github.com/apache/shardingsphere/issues).
- Read documentation: [ShardingSphere Doc](https://shardingsphere.apache.org/document/current/en/overview).

Please pay attention on issues you submitted, because we maybe need more details. 
If no response anymore and we cannot reproduce it on current information, we will **close it**.

Please answer these questions before submitting your issue. Thanks!

### Which version of ShardingSphere did you use?
5.0.0,  master

### Which project did you use? ShardingSphere-JDBC or ShardingSphere-Proxy?

### Expected behavior
 throw new IllegalStateException ...

### Actual behavior
return 

### Reason analyze (If you can)
when i write a wrong jdbc url ( mistake ),  the error log is not good .
log like this 

Caused by: java.lang.IllegalStateException: Database type inconsistent with 'org.apache.shardingsphere.infra.database.type.dialect.MySQLDatabaseType@1da53f4f' and 'null'
	at com.google.common.base.Preconditions.checkState(Preconditions.java:826) ~[guava-30.1.1-jre.jar:?]
	at org.apache.shardingsphere.infra.database.type.DatabaseTypeRecognizer.getDatabaseType(DatabaseTypeRecognizer.java:45) ~[shardingsphere-infra-common-5.0.0.jar:5.0.0]
	at org.apache.shardingsphere.infra.rule.builder.schema.SchemaRulesBuilder.buildRules(SchemaRulesBuilder.java:69) ~[shardingsphere-infra-common-5.0.0.jar:5.0.0]
	at org.apache.shardingsphere.mode.manager.memory.MemoryContextManagerBuilder.build(MemoryContextManagerBuilder.java:53) ~[shardingsphere-memory-mode-core-5.0.0.jar:5.0.0]
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.createContextManager(ShardingSphereDataSource.java:75) ~[shardingsphere-jdbc-core-5.0.0.jar:5.0.0]
	at org.apache.shardingsphere.driver.jdbc.core.datasource.ShardingSphereDataSource.<init>(ShardingSphereDataSource.java:61) ~[shardingsphere-jdbc-core-5.0.0.jar:5.0.0]
	at org.apache.shardingsphere.driver.api.ShardingSphereDataSourceFactory.createDataSource(ShardingSphereDataSourceFactory.java:79) ~[shardingsphere-jdbc-core-5.0.0.jar:5.0.0]
	

### Steps to reproduce the behavior, such as: SQL to execute, sharding rule configuration, when exception occur etc.

### Example codes for reproduce this issue (such as a github link).

--- a/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeRecognizer.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/main/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeRecognizer.java
@@ -48,12 +48,11 @@ public final class DatabaseTypeRecognizer {
         return null == result ? DatabaseTypeRegistry.getDefaultDatabaseType() : result;
     }
     
-    @SuppressWarnings("ReturnOfNull")
     private static DatabaseType getDatabaseType(final DataSource dataSource) {
         try (Connection connection = dataSource.getConnection()) {
             return DatabaseTypeRegistry.getDatabaseTypeByURL(connection.getMetaData().getURL());
         } catch (final SQLException ex) {
-            return null;
+            throw new IllegalArgumentException("An error occurred when open datasource connection!", ex);
         }
     }
 }

--- a/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeRecognizerTest.java
+++ b/shardingsphere-infra/shardingsphere-infra-common/src/test/java/org/apache/shardingsphere/infra/database/type/DatabaseTypeRecognizerTest.java
@@ -104,6 +104,13 @@ public final class DatabaseTypeRecognizerTest {
         DatabaseTypeRecognizer.getDatabaseType(differentDataSources);
     }
 
+    @Test(expected = IllegalArgumentException.class)
+    public void assertCantGetConnectionFromDataSource() throws SQLException {
+        DataSource mockDataSource = mock(DataSource.class);
+        when(mockDataSource.getConnection()).thenThrow(SQLException.class);
+        DatabaseTypeRecognizer.getDatabaseType(Arrays.asList(mockDataSource));
+    }
+
     private DataSource mockDataSource(final DatabaseType databaseType) throws SQLException {
         DataSource result = mock(DataSource.class);
         Connection connection = mock(Connection.class, RETURNS_DEEP_STUBS);

